snowshu/adapters/source_adapters/snowflake_adapter.py

```diff
@@ -103,6 +103,100 @@ def _get_all_schemas(self, database: str, exclude_defaults: Optional[bool] = Fal
             f'Done. Found {len(schemas)} schemas in {database} database.')
         return schemas
 
+    def _get_all_tables(self, database: str, schema: str) -> List[str]:
+        database = self.quoted(database)
+        schema = self.quoted(schema)
+        logger.debug(f'Collecting tables from {schema} schema in {database} database in snowflake...')
+        show_result = self._safe_query(f'SHOW TERSE TABLES IN SCHEMA {database}.{schema}')['name'].tolist()
+        tables = list(set(show_result))
+        logger.debug(f'Done. Found {len(tables)} tables in {schema} schema of {database} database.')
+        return tables
+
+    def generate_schema(self, name: str, database: str = 'SNOWSHU'):
+        """Create a schema in the specified database.
+
+            Args:
+                name: The name of the schema to create.
+                database: The database where the schema will be created.
+                          Defaults to 'SNOWSHU'.
+        """
+        corrected_database, corrected_name = (
+            self._correct_case(x) for x in (database, name))
+        try:
+            logger.debug("Creating a schema %s in %s database...",
+                         corrected_name, corrected_database)
+            query = f'''CREATE SCHEMA IF NOT EXISTS
+                    {corrected_database}.{corrected_name}'''
+            result = self._safe_query(query)
+            logger.info("Schema creation result: %s", result['status'][0])
+        except ValueError as err:
+            error_message = (
+                f"An error occurred while creating the schema {corrected_name} "
+                f"in database {corrected_database}: {err}"
+            )
+            logger.error(error_message)
```

Comment:
Does it makes sense to continue execution(no re-raise here) if CREATE SCHEMA statement wasn't executed? 
Same concern for all functions introduced in this PR

Reply:
Yeah, good observation I might have missed that..